tv_apps/smart-system/js/app_install_manager.js

```diff
@@ -232,35 +232,68 @@ var AppInstallManager = {
     var dialogConfig;
 
     if (unrecoverable) {
-      dialogConfig = {
-        type: 'unrecoverable',
-        title: 'unrecoverable-error-title',
-        body: 'unrecoverable-error-body',
-        confirm: {
-          title: 'unrecoverable-error-action',
-          cb: () => { this.dispatchResponse(id, 'webapps-uninstall-granted'); }
-        }
-      };
+      this.hookSimpleNavigator(
+        [this.appUninstallConfirmButton],
+        this.appUninstallConfirmButton);
+
+      this.appUninstallDialog.classList.add('visible');
+
+      // Hide Cancel button and adjust its position.
+      this.appUninstallCancelButton.style.display = 'none';
+      this.appUninstallConfirmButton.style.marginLeft = '0';
+      this.appUninstallConfirmButton.parentNode.setAttribute('data-items', 1);
+
+      navigator.mozL10n.setAttributes(this.appUninstallMessage,
+        'unrecoverable-error-body'
+      );
     } else {
-      var nameObj = { name: manifest.name };
-      dialogConfig = {
-        type: 'remove',
-        title: {id: 'delete-title', args: nameObj},
-        body: {id: 'delete-body', args: nameObj},
-        cancel: {
-          title: 'cancel',
-          cb: () => { this.dispatchResponse(id, 'webapps-uninstall-denied'); }
-        },
-        confirm: {
-          title: 'delete',
-          type: 'danger',
-          cb: () => { this.dispatchResponse(id, 'webapps-uninstall-granted'); }
-        }
-      };
-    }
-
-    var dialog = new ConfirmDialogHelper(dialogConfig);
-    dialog.show(document.body);
+      this.hookSimpleNavigator(
+        [this.appUninstallCancelButton, this.appUninstallConfirmButton],
+        this.appUninstallCancelButton);
+
+      this.appUninstallDialog.classList.add('visible');
+
+      // Show Cancel button.
+      this.appUninstallCancelButton.style.display = '';
+      this.appUninstallConfirmButton.style.marginLeft = '';
+      this.appUninstallConfirmButton.parentNode.setAttribute('data-items', 2);
+
+      navigator.mozL10n.setAttributes(this.appUninstallMessage,
+        'delete-body', {'name': manifest.name}
+      );
+    }
+
+    this.uninstallCallback = (function ai_uninstallCallback() {
+      this.unhookSimpleNavigator();
+      this.dispatchResponse(id, 'webapps-uninstall-granted');
+    }).bind(this);
+
+    this.uninstallCancelCallback = (function ai_cancelCallback() {
+      this.unhookSimpleNavigator();
+      this.dispatchResponse(id, 'webapps-uninstall-denied');
+    }).bind(this);
+  },
+
+  handleUninstallDialog: function ai_handleUninstallDialog(evt) {
+    if (evt) {
+      evt.preventDefault();
```

Comment:
I don't think we need "preventDefault" since we already switched to smart-button.